src/client/common/application/debugSessionTelemetry.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import { inject, injectable } from 'inversify';
+import { DebugAdapterTracker, DebugAdapterTrackerFactory, DebugSession, Event, EventEmitter, ProviderResult } from 'vscode';
+import { DebugProtocol } from 'vscode-debugprotocol';
+
+import { IExtensionSingleActivationService } from '../../activation/types';
+import { AttachRequestArguments, ConsoleType, LaunchRequestArguments, TriggerType } from '../../debugger/types';
+import { sendTelemetryEvent } from '../../telemetry';
+import { EventName } from '../../telemetry/constants';
+import { IDisposableRegistry } from '../types';
+import { StopWatch } from '../utils/stopWatch';
+import { IDebugService } from './types';
+
+class TelemetryTracker implements DebugAdapterTracker {
+    private timer = new StopWatch();
+    private trigger: TriggerType = 'launch';
```

Comment:
```suggestion
    private readonly trigger: TriggerType = 'launch';
```
Personally I prefer to see `readonly` in properties, easy to tell whether they are updated or not (readonly = only read and initialized in `ctor`).

Reply:
Old C# habit. Readonly tells me at a glance that they are `read only` 😄

Reply:
Sure will do. Didn't realize I could set them twice if they were read only.